### ISSUE TITLE
[Feature Request] Enable Pulling of Episode Count from Kitsu.io

### ISSUE THREAD

**Issue by dpearson2699** [SuperMarcus/NineAnimator#278]
Issue:
I track the shows I've watched using Kitsu.io. I have the app both on my Mac and iPhone. So NineAnimator successfully pushes the correct episode to Kitsu.io but it doesn't seem like the app pulls from Kitsu.io to show the correct episode I am on. Ex: I watch 2 episodes on my iPhone, but my Mac doesn't show that progress. Even if I go into Library/Kitsu.io/Currently Watching. It'll say that I'm two episodes behind. But when I check the actual website it shows the correct episode I'm on. This is quite annoying when switching between devices trying to figure out what episode you are on.

Solution:
Enable through the Kitsu.io API the ability to pull the episode count the user is on, instead of checking the local storage to see the most recently watched episode. This will allow you to switch between devices and continue where you left off.

**Comment by Layendan** [comment]
I think that this feature would be a great idea. I think that we should also implement it for MAL, Anilist, and Simkl. The problem with this is the logistics of it. The app is definitely able to pull from those sites, but it can't add the episodes to recently watched the way the app currently works. To add to recently watched, the app needs a source anime (eg. watched from gogo) but third-party anime lists return non-source animes (eg. Kitsu or MAL) because the anime doesn't have a source with it, the app cannot add it to your recently watched.

Now it's always possible to make it so that the app searches the currently selected source and automatically chooses the anime with the closest name to it, or sync progress using accounts; however, I do not think that the developers of this app intend to add those features any time soon.

**Comment by dpearson2699** [comment]
> The app is definitely able to pull from those sites, but it can't add the episodes to recently watched the way the app currently works.

Um, maybe I’m misunderstanding what you’re saying but the app can definitely add the most recently watched episode to those sites. Actually it does that now. For example if I watch episode 139 of Bleach, once I finish it and I look at Kitsu.io, it will show that I just watched that episode (I do have the anime in my watching now list. Not sure if that changes anything). Now I know there is a bug with watching dubbed anime and tracking when it relates to MAL or Anilist (see issue #242), but Kitsu doesn’t have that issue.

**Comment by Layendan** [comment]
What I meant by this was that the app can push data to those sites (you watch an episode on an app and it updates kitsu [app already does that]) and it CAN pull from those sites (added watched episode and the app can read this [not implemented]) but the app cannot add to recently watched pulled data from those sites. The algorithm that the developers have made are one way (episode watched on app sent to anime list but not the way around). Because the app needs a source (gogo, 9anime...) to add to recently watched and that the algorithm is only one way, it's not possible to retrieve anime list episodes and add them to recently watched.

Sorry if I was being vague, I hope this clears things up.

I also want this feature in the app, but implementing this will most likely take up too much time and resources.

tldr: The app has to be rewritten a lot to add this feature.

**Comment by BEEDELLROKEJULIANLOCKHART** [comment]
[Tachiyomi](https://github.com/tachiyomiorg/tachiyomi) has solved this problem for manga. Duplicating how it performs this might be useful.
